polling/src/main/java/io/atleon/polling/reactive/PollingEventLoop.java

```diff
@@ -0,0 +1,109 @@
+package io.atleon.polling.reactive;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.SignalType;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Scheduler;
+import reactor.util.annotation.NonNull;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class PollingEventLoop<K extends PollingRecord<V>, V> implements Sinks.EmitFailureHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(PollingEventLoop.class);
+    private final Scheduler scheduler;
+    private final Poller<K, V> poller;
+    private final Sinks.Many<Collection<K>> sink;
+    private final PollingEvent pollEvent;
+    private final AtomicBoolean active;
+    private final Duration pollingInterval;
+
+    public PollingEventLoop(final Scheduler scheduler,
+                            final Poller<K, V> poller,
+                            final Duration pollingInterval,
+                            final Sinks.Many<Collection<K>> sink) {
+        this.scheduler = scheduler;
+        this.poller = poller;
+        this.pollingInterval = pollingInterval;
+        this.sink = sink;
+        this.active = new AtomicBoolean(true);
+        this.pollEvent = new PollingEvent();
+    }
+
+    void onRequest(long toAdd) {
+        active.set(true);
+        pollEvent.scheduleImmediate();
+    }
+
+    public Mono<Void> stop() {
+        return Mono
+                .defer(() -> {
+                    pollEvent.stop();
+                    return Mono.<Void>empty();
+                })
+                .onErrorResume(e -> Mono.empty());
```

Comment:
I think you could simplify this even further:
```java
return Mono.fromRunnable(pollEvent::stop)
    .onErrorResume(error -> Mono.empty());
```

Reply:
Agreed, I'll update this.